lib/Runtime/Library/JavascriptRegularExpression.cpp

```diff
@@ -870,21 +870,6 @@ namespace Js
 
     Var JavascriptRegExp::CallExec(RecyclableObject* thisObj, JavascriptString* string, PCWSTR varName, ScriptContext* scriptContext)
     {
-        JavascriptRegExp* regExObj = nullptr;
-
-        // TODO: The built-in "exec" in ES6 is supposed to access the RegExp flags. Since the flags
-        // can be overridden and return a different value than what's passed to the constructor,
-        // the pattern needs to be recompiled. However, the way the observability check is currently
-        // implemented, it reduces the perf quite a lot.
-        //
-        // Therefore, we recompile the pattern here before calling "exec" for the time being, but
-        // this will be moved to "exec".
-        if (JavascriptRegExp::Is(thisObj))
-        {
-            regExObj = JavascriptRegExp::FromVar(thisObj);
-            regExObj->RecompilePatternForExecIfNeeded(scriptContext);
```

Comment:
This was the only place it was referenced, so it is now removed with commit 97f7269d2e40b69c33da24f7ff9df51805b288e2. I did the same for [`JavascriptRegExp::HasObservableStickyFlag`](https://github.com/Microsoft/ChakraCore/pull/813/commits/97f7269d2e40b69c33da24f7ff9df51805b288e2#diff-8dbc33a433cf59be6a51b9a3318ad7d2L71).